sync/sync.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd ) ; readonly dir
+cd "${dir}/.."
+
+# Stage 1 sync
+set -x
+vendir sync
+{ set +x; } 2>/dev/null
+
+# Remove trailing whitespace end of lines (hack to fix vendir bug)
+find vendor/ -type f -exec sed -i 's/[[:space:]]*$//' {} \;
+
+# Patches
+./sync/patches/network-policies/patch.sh
+./sync/patches/deployment/patch.sh
+./sync/patches/crds/patch.sh
+./sync/patches/vpa/patch.sh
+./sync/patches/helper/patch.sh
+./sync/patches/values/patch.sh
+
+HELM_BASE="helm/external-dns-app"
+VENDIR_BASE="vendor/external-dns"
+
+# HELM_DOCS="docker run --rm -u $(id -u) -v ${PWD}:/helm-docs -w /helm-docs jnorwood/helm-docs:v1.11.0"
+# $HELM_DOCS --template-files=sync/readme.gotmpl -g helm/envoy-gateway -f values.yaml -o README.md
```

Comment:
in ingress-nginx-app I installed helm-docs in the github actions yaml
see https://github.com/giantswarm/ingress-nginx-app/blob/main/.github/workflows/validate-sync-diffs.yaml#L23

Reply:
thanks, I'll add that to envoy-gateway sync script. for now leaving it out from this repo as we don't use helm-docs at this stage. we can add it in a second step.